contracts/interfaces/IQuoterWithGas.sol

```diff
@@ -0,0 +1,93 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity >=0.7.5;
+pragma abicoder v2;
+
+/// @title Quoter Interface
+/// @notice Supports quoting the calculated amounts from exact input or exact output swaps
+/// @notice For each pool also tells you the number of initialized ticks crossed and the sqrt price of the pool after the swap.
+/// @notice Also provides an estimate of the gas used by the swap.
+/// @dev These functions are not marked view because they rely on calling non-view functions and reverting
+/// to compute the result. They are also not gas efficient and should not be called on-chain.
+interface IQuoterWithGas {
```

Comment:
i'm kinda anti using this, would rather delete, at least in part b/c the implementation includes the gas used in computing how many initialized ticks were crossed

Reply:
Moved to within Quoter. As discussed we may end up not using it but lets see